src/VirtualClient/VirtualClient.Actions/Kafka/KafkaServerExecutor.cs

```diff
@@ -0,0 +1,498 @@
+                {
+                    if (!cancellationToken.IsCancellationRequested)
+                    {
+                        await this.LogProcessDetailsAsync(process, telemetryContext, "ClusterId");
+                        process.ThrowIfWorkloadFailed();
+                    }
+
+                    output = process.StandardOutput.ToString();
+                    telemetryContext.AddContext("Cluster Id", output);
+                }
+
+                return output;
+            }
+            catch (OperationCanceledException)
+            {
+                // Expected whenever certain operations (e.g. Task.Delay) are cancelled.
+                return null;
+            }
+            catch (Exception exc)
+            {
+                this.Logger.LogMessage(
+                    $"{this.TypeName}.GetClusterIdError",
+                    LogLevel.Error,
+                    telemetryContext.Clone().AddError(exc));
+
+                throw;
+            }
+        }
+
+        private Task StartServerAndWaitForExitAsync(string logMessage, string command, string commandArguments, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken, string toolName = null, bool addCleanupTasks = true)
+        {
+            return (this.ServerRetryPolicy ?? Policy.NoOpAsync()).ExecuteAsync(async () =>
+            {
+                try
+                {
+                    telemetryContext.AddContext(nameof(commandArguments), commandArguments);
+                    using (IProcessProxy process = await this.RunProcessAsync(command, commandArguments, workingDirectory, telemetryContext, cancellationToken, runElevated: true))
+                    {
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            ConsoleLogger.Default.LogMessage(logMessage, telemetryContext);
+                            await this.LogProcessDetailsAsync(process, telemetryContext, toolName);
+                            process.ThrowIfWorkloadFailed();
+                        }
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                    // Expected whenever certain operations (e.g. Task.Delay) are cancelled.
+                }
+                catch (Exception exc)
+                {
+                    this.Logger.LogMessage(
+                        $"{this.TypeName}.StartServerAndWaitForExitAsyncError",
+                        LogLevel.Error,
+                        telemetryContext.Clone().AddError(exc));
+
+                    throw;
+                }
+            });
+        }
+
+        private async Task ConfigurePropertiesFileAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string logDir = "/tmp";
```

Comment:
Does this PR have Linux support or not? Just to make sure we are coding for both OS in the future

Reply:
Current PR targets support for windows. Linux will be tested and raised as separate PR.